<launch>

    <!-- Intel RealSense ROS Node - Cam with ID 828112071570 -->
    <include file="$(find realsense2_camera)/launch/rs_camera.launch"> 
        <arg name="camera" value="cam_1"/>
        <arg name="serial_no" value="828112071570"/>
        <arg name="filters" value="spatial,temporal,pointcloud"/>
    </include> 

    <!-- Intel RealSense ROS Node - Cam with ID 843112072743 -->
    <include file="$(find realsense2_camera)/launch/rs_camera.launch"> 
        <arg name="camera" value="cam_2"/>
        <arg name="serial_no" value="843112072743"/>
        <arg name="filters" value="spatial,temporal,pointcloud"/>
    </include>  

    <!-- Extract two pointclouds directly from the Pointcloud topics -->

    <!-- Apply Passthrough filter with Z Distance of 2.2 on cam_1 -->

    <!-- Apply Passthrough filter with Z Distance of 2 on cam_2 -->

    <!-- Apply Outlier filter on both camera images -->

    <!-- Perform initial transformation X-rotation -45° - Z-rotation 180° on cam_1 -->
    
    <!-- Perform nonlinear ICP and extract rotation matrix -->
    
    <!-- Perform coordinate transform from cam_2_link relative to world frame -->
    <node type="static_transform_publisher" name="static_transform_cam_2_cam_1" pkg="tf" args="-0.925 0.1563 3.022 -0.877438110987 0.0948478796321 0.46937536655 0.028160369202 world_frame cam_2_link 9" />
    <!-- Perform coordinate transform from cam_1_link relative to world frame -->
    <node type="static_transform_publisher" name="static_transform_world_cam_1" pkg="tf" args="1.0 0.0 2.9 0.0311474473239 -0.891296151451 0.0487023250985 -0.449721125214 world_frame cam_1_link 9" />
    <!-- Start rviz with predefined config: added both Intel Pointcloud Streams and set world_frame as frame -->
    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find perception)/config/2cam_pointcloud.rviz" />
</launch>
